src/commonMain/kotlin/exchange.dydx.abacus/state/manager/V4StateManagerAdaptor.kt

```diff
@@ -1034,6 +1034,100 @@ class V4StateManagerAdaptor(
         return payload
     }
 
+    override fun commitTriggerOrders(callback: TransactionCallback): List<HumanReadablePlaceOrderPayload>? {
+        val payloads = triggerOrdersPayload()
+
+        payloads.cancelOrderPayloads.forEach {
+            val string = Json.encodeToString(it)
+            val analyticsPayload = analyticsUtils.formatCancelOrderPayload(it)
+
+            val uiClickTimeMs = Clock.System.now().toEpochMilliseconds().toDouble()
+            tracking(AnalyticsEvent.TradeCancelOrderClick.rawValue, analyticsPayload)
+
+            val transactionCallback =
+                { response: String?, uiDelayTimeMs: Double, submitTimeMs: Double ->
+                    val error = parseTransactionResponse(response)
+                    if (error == null) {
+                        tracking(
+                            AnalyticsEvent.TradeCancelOrder.rawValue,
+                            ParsingHelper.merge(uiTrackingParams(uiDelayTimeMs), analyticsPayload)
+                                ?.toIMap(),
+                        )
+                        ioImplementations.threading?.async(ThreadingType.abacus) {
+                            this.orderCanceled(it.orderId)
+                            this.cancelOrderRecords.add(
+                                CancelOrderRecord(
+                                    subaccountNumber,
+                                    it.clientId,
+                                    submitTimeMs,
+                                ),
+                            )
+                        }
+                    }
+                    send(error, callback, it)
```

Comment:
I feel like it's sort of an antipattern to be calling `send` multiple times in `commitTriggerOrders` (it's being called once for every order action, and then once at the end, to signal a success if no actions were taken). This makes for some weird FE ui too (if I'm waiting for onSuccess) - but unsure if there's a better solution here that doesn't involve making a new `TransactionType` which I'd really rather not do

Reply:
umm maybe @johnqh can chime in here too. im not sure what the timeline is for supporting replacing orders / sending / canceling multiple orders in one tx, i think that would solve most of those concerns